SSA/Projects/FullyHomomorphicEncryption/Rewrites.lean

```diff
@@ -72,3 +72,79 @@ noncomputable def p1 : PeepholeRewrite (Op q n) [.polynomialLike, .polynomialLik
     }
 
 end ExampleComm
+
+section ExampleModulo
+
+@[irreducible]
+def irreduciblePow (q n : Nat) : Nat := q^n
+infixl:50 "**" => irreduciblePow
+
+
+variable {q : Nat} {n : Nat} [hq : Fact (q > 1)]
+
+-- code generator does not support recursor 'Decidable.rec' yet, consider using 'match ... with' and/or structural recursion
+noncomputable def lhs := [fhe_com q, n, hq| {
+^bb0(%a : ! R):
+  %one_int = "arith.const" () {value = 1}: () -> (i16)
+  %two_to_the_n = "arith.const" () {value = $((2**n : Int))}: () -> (index)
+  %x2n = "poly.monomial" (%one_int,%two_to_the_n) : (i16, index) -> (! R)
+  %oner = "poly.const" () {value = 1}: () -> (! R)
+  %p = "poly.add" (%x2n, %oner) : (! R, ! R) -> (! R)
+  %v1 = "poly.add" (%a, %p) : (! R, ! R) -> (! R)
+  "return" (%v1) : (! R) -> ()
+}]
+
+def rhs := [fhe_com q, n, hq | {
+^bb0(%a : ! R):
+  "return" (%a) : (! R) -> ()
+}]
+
+/- 'lhs' depends on axioms: [propext, Quot.sound, Classical.choice, ROfZComputable_stuck_term] -/
+#print axioms lhs
```

Comment:
```suggestion
/--info: 'lhs' depends on axioms: [propext, Quot.sound, Classical.choice, ROfZComputable_stuck_term] -/
#guard_msgs in #print axioms lhs
```

This way, `lake build` is less noisy. Plus, we wouldn't have to manually compare axioms anymore; a change in axioms would become a build/CI error